Assets/MRTK/SDK/Features/Utilities/Solvers/HandConstraint.cs

```diff
@@ -328,15 +333,35 @@ protected virtual Vector3 CalculateGoalPosition()
             Bounds trackedHandBounds;
 
             if (trackedController != null &&
-                handBounds.Bounds.TryGetValue(trackedController.ControllerHandedness, out trackedHandBounds))
+                handBounds.LocalBounds.TryGetValue(trackedController.ControllerHandedness, out trackedHandBounds))
             {
+                MixedRealityPose? palmPose = GetPalmPose(trackedController);
+
+                // If we somehow were unable to obtain a palm pose, we just quit;
+                // we require a valid palm pose to perform the hand-space transformations.
+                if (palmPose.HasValue == false)
+                {
+                    return goalPosition;
+                }
```

Comment:
In this case, is the returned `goalPosition` "good enough" to use this solver with non-jointed input sources? We've had requests to convert things like the hand menu to be usable with controllers, which would likely need some additional work for synthesizing the pose being detected (palm up, etc). Wondering if you have any thoughts on how far just this would get us!

Reply:
@keveleigh The rest of the code here would also fail if you couldn't obtain joints from the hand. So yep, this wouldn't work with controllers; but the rest of this entire hand constraint system also _assumes_ we are only using this with hands. I agree, it would be valuable to be able to use this system with non-hand controllers; but that would be better for another PR, imo.

Reply:
Oh yeah, definitely! Agreed that it's outside the scope of this PR. Mostly wanted to get your take on it, since I haven't gone too deep into `HandConstraint` myself.